add draft of week-4 workshop (#111)
CodeYourFuture/CYF-Workshops: time-string/index.md

# `formatSecondsAsString`

## Learning Objectives

```objectives
- [ ] Break a down a problem into smaller cases
- [ ] Write tests to check logic for smaller cases
```

> When programming, you will frequently encounter large problems. As a developer, it's essential that we can break down complex problems into smaller problems. In this workshop, we'll use tests to break down the implementation of `formatAsSecondsAsString`.

Reviewer: ```suggestion
When programming, you will often encounter large problems. Developers must learn to break down complex problems into smaller problems. In this workshop, we'll use tests to break down the implementation of `formatAsSecondsAsString`.
```

> Given a time in seconds, 

Reviewer: ```suggestion
_Given_ a time in seconds, 
```

> When `formatAsSecondsAsString` is called with the time

Reviewer: ```suggestion
_When_ `formatAsSecondsAsString` is called with the time
```

> Then we it should return a well formatted time string.

Reviewer: ```suggestion
_Then_ we it should return a well formatted time string.
```

For example, if we call `formatAsSecondsAsString` with 390
then we should get "6 minutes and 30 seconds"

If we call, `formatAsSecondsAsString` with `6327894` then we should get ""

### 🧩 Break down the problem

(10 mins)

To break down this problem, we will only start considering seconds inputs <= 60. Why do you think this is a good place to start?

```js
formatAsSecondsAsString(1) // "1 second" 
```

```js
formatAsSecondsAsString(50) // "50 seconds" 
```

> Spend 10 minutes, think about some other test cases you might wish to explore to check this functionality

Reviewer: ```suggestion
Spend 10 minutes, think about some other test cases you can explore to check this functionality
```

### 🧪 Set up tests

(10 mins)

In this directory (`time-string`), create a `package.json` and install Jest as you did in the prep material.

> Once you're set up create a test file, `time-string.test.js`. In there, write your first test.

Reviewer: ```suggestion
Next, create a test file, `time-string.test.js`. In there, write your first test.
```
